subsys/bluetooth/host/hci_core.c

```diff
@@ -4890,6 +4912,138 @@ static void le_adv_ext_report(struct net_buf *buf)
 		net_buf_pull(buf, evt->length);
 	}
 }
+
+#if defined(CONFIG_BT_PER_ADV_SYNC)
+
+static struct bt_le_per_adv_sync *get_per_adv_sync(uint16_t handle)
+{
+	for (int i = 0; i < ARRAY_SIZE(per_adv_sync_pool); i++) {
+		if (per_adv_sync_pool[i].handle == handle &&
+		    atomic_test_bit(per_adv_sync_pool[i].flags,
+				     BT_PER_ADV_SYNC_SYNCED)) {
+			return &per_adv_sync_pool[i];
+		}
+	}
+
+	return NULL;
+}
+
+static void le_per_adv_report(struct net_buf *buf)
+{
+	struct bt_hci_evt_le_per_advertising_report *evt;
+	struct bt_le_per_adv_sync *per_adv_sync;
+	struct bt_le_per_adv_sync_recv_info info;
+
+	if (buf->len < sizeof(*evt)) {
+		BT_ERR("Unexpected end of buffer");
+		return;
+	}
+
+	evt = net_buf_pull_mem(buf, sizeof(*evt));
+
+	per_adv_sync = get_per_adv_sync(evt->handle);
+
+	if (!per_adv_sync) {
+		BT_ERR("Unknown handle 0x%04X for per adv report", evt->handle);
+		return;
+	}
+
+	info.tx_power = evt->tx_power;
+	info.rssi = evt->rssi;
+	info.cte = evt->cte;
+	info.addr = &per_adv_sync->params.addr;
+
+	if (per_adv_sync->cb && per_adv_sync->cb->recv) {
+		per_adv_sync->cb->recv(per_adv_sync, &info, &buf->b);
+	}
+}
+
+static void le_per_adv_sync_established(struct net_buf *buf)
+{
+	struct bt_hci_evt_le_per_adv_sync_established *evt =
+		(struct bt_hci_evt_le_per_adv_sync_established *)buf->data;
+	struct bt_le_per_adv_sync_synced_info sync_info;
+
+	if (buf->len != sizeof(*evt)) {
+		BT_ERR("Unexpected event size: Was %u expected %u",
+		       buf->len, (uint32_t)sizeof(*evt));
+		return;
+	}
+
+	if (!syncing_per_adv_sync ||
+	    syncing_per_adv_sync->params.sid != evt->sid ||
+	    bt_addr_le_cmp(&syncing_per_adv_sync->params.addr,
+			   &evt->adv_addr)) {
+		BT_ERR("Unexpected per adv sync established event");
+		return;
+	}
+
+	atomic_set_bit(syncing_per_adv_sync->flags, BT_PER_ADV_SYNC_SYNCED);
+	if (atomic_test_bit(syncing_per_adv_sync->flags,
```

Comment:
Did you consider the case where the application starts the scanner while this is happening?
In that case I think EXPLICIT_SCAN would be set, and we should not disable scan in that case.

If that use-case is not supported I think we need to add some handling to bt_le_scan_start to prevent the application from starting the scanner while establishing sync.

Reply:
Hmm, yeah it's actually somewhat complicated with create_conn, per_adv_sync_create and explicit scan now. 

Currently I've modifeid sync create to return EBUSY if explicit scan is enabled, and if scan is explicitely enabled, then `BT_PER_ADV_SYNC_SCAN_INITIATED` will be cleared, and won't attempt to disable scan on sync. 

There's some coordination between create sync and create connection that I hope works with my update, which should also do so that if explicit scan is disabled, should restart the scan for per adv sync. This is done by hooking onto the `bt_le_scan_update` function.